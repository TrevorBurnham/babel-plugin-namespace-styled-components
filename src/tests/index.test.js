import pluginTester from 'babel-plugin-tester';
import path from 'path';
import plugin from '../index';

pluginTester({
  plugin,
  pluginName: 'namespace-styled-components',
  pluginOptions: {
    namespace: '.namespace',
  },
  filename: __filename,
  snapshot: true,
  tests: [
    {
      title: 'namespaces a style block with no selectors',
      fixture: path.join(__dirname, 'fixtures/simple.js'),
    },

    {
      title: 'namespaces a style block with &&',
      fixture: path.join(__dirname, 'fixtures/doubleAmpersand.js'),
    },

    {
      title: 'namespaces a style block with a sibling selector',
      fixture: path.join(__dirname, 'fixtures/siblingSelector.js'),
    },

    {
      title: 'namespaces a style block with interpolated selectors',
      fixture: path.join(__dirname, 'fixtures/interpolatedSelector.js'),
    },

    {
      title: 'namespaces a style block with interpolated mixins',
      fixture: path.join(__dirname, 'fixtures/interpolatedMixin.js'),
    },

    {
      title: 'namespaces a style block with interpolations before and after a declaration',
      fixture: path.join(__dirname, 'fixtures/interpolationSandwich.js'),
    },

    {
      title: 'namespaces a style block with a media query',
      fixture: path.join(__dirname, 'fixtures/mediaQuery.js'),
    },

    {
      title: 'does not namespace style blocks in helpers',
      fixture: path.join(__dirname, 'fixtures/helpers.js'),
      snapshot: false,
    },

    {
      title: 'throws an error if no namespace is defined',
      fixture: path.join(__dirname, 'fixtures/simple.js'),
      pluginOptions: {
        namespace: undefined,
      },
      snapshot: false,
      error: /option `namespace` must be provided/,
    },

    {
      title: 'throws an error if the styles canâ€™t be parsed',
      fixture: path.join(__dirname, 'fixtures/malformatted.js'),
      snapshot: false,
      error: /Unexpected '\/'/,
    },
  ],
});
