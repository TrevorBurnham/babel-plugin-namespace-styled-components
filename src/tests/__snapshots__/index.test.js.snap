// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`namespace-styled-components namespaces a style block with &&: namespaces a style block with && 1`] = `
"
import styled from 'styled-components';

export default styled.input\`
  && {
    border: \${props => props.borderWidth} solid \${props => props.borderColor};
  }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';

export default styled.input\`.namespace && {
    border: \${props => props.borderWidth} solid \${props => props.borderColor};
  }\`;
"
`;

exports[`namespace-styled-components namespaces a style block with a sibling selector: namespaces a style block with a sibling selector 1`] = `
"
import styled from 'styled-components';

export default styled.button\`
  padding: \${props => props.padding};

  & + & {
    margin-left: \${props => props.spaceBetween};
  }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';

export default styled.button\`.namespace & { 
  padding: \${props => props.padding};
 }

  .namespace & + & {
    margin-left: \${props => props.spaceBetween};
  }\`;
"
`;

exports[`namespace-styled-components namespaces a style block with interpolated mixins: namespaces a style block with interpolated mixins 1`] = `
"
import styled, { css } from 'styled-components';

const displayBlockMixin = css\`
  display: block;
\`;

export default styled.div\`
  \${displayBlockMixin}

  position: relative;

  &:hover {
    opacity: 1;
  }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

import styled, { css } from 'styled-components';

const displayBlockMixin = css\`
  display: block;
\`;

export default styled.div\`.namespace & { 
  \${displayBlockMixin}

  position: relative;
 }

  .namespace &:hover {
    opacity: 1;
  }\`;
"
`;

exports[`namespace-styled-components namespaces a style block with interpolated selectors: namespaces a style block with interpolated selectors 1`] = `
"
import styled from 'styled-components';

const Child = styled.span;

export default styled.div\`
  position: relative;

  \${Child} {
    \${props => props.childStyles};
  }

  \${Child} + &,
  & + \${Child} {
    margin-right: \${props => props.spaceBetween};
  }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';

const Child = styled.span;

export default styled.div\`.namespace & { 
  position: relative;
 }

  \${Child}

  .namespace & {
    \${props => props.childStyles};
  }

  \${Child}

  .namespace + &,
  .namespace & + \${Child} {
    margin-right: \${props => props.spaceBetween};
  }\`;
"
`;

exports[`namespace-styled-components namespaces a style block with interpolations before and after a declaration: namespaces a style block with interpolations before and after a declaration 1`] = `
"
import styled from 'styled-components'

const stringify = (key, val) => \`\${key}: \${val};\`

export default styled.span\`
  && {
    \${() => stringify('background-color', 'red')};

    \${() => stringify('font-size', '12px')};

    color: pink;

    \${() => stringify('display', 'flex')};

    &:hover {
      color: red;
    }
  }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';

const stringify = (key, val) => \`\${key}: \${val};\`;

export default styled.span\`.namespace && {
    \${() => stringify('background-color', 'red')};

    \${() => stringify('font-size', '12px')};

    color: pink;

    \${() => stringify('display', 'flex')}
  };

    .namespace &&:hover {
      color: red;
    }\`;
"
`;

exports[`namespace-styled-components namespaces a style block with no selectors: namespaces a style block with no selectors 1`] = `
"
import styled from 'styled-components';

export default styled.div\`
  background-color: \${props => props.backgroundColor};
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components';

export default styled.div\`.namespace & { 
  background-color: \${props => props.backgroundColor};
 }\`;
"
`;
